tools/java/org/unicode/cldr/util/CLDRConfig.java

```diff
@@ -533,15 +533,37 @@ public int getProperty(String k, int defVal) {
         }
     }
 
-    public File getCldrBaseDirectory() {
-        String dir = getProperty("CLDR_DIR", null);
-        if (dir != null) {
-            return new File(dir);
-        } else {
-            return null;
+    static class FileWrapper {
+        private File cldrDir = null;
+        private FileWrapper() {
+            String dir = getInstance().getProperty("CLDR_DIR", null);
+            if (dir != null) {
+                cldrDir = new File(dir);
+            } else {
+                cldrDir = null;
+            }
+        }
+        public File getCldrDir() {
+            return this.cldrDir;
+        }
+        // singleton
+        private static FileWrapper fileWrapperInstance = null;
+        public static FileWrapper getFileWrapperInstance() {
+            if (fileWrapperInstance == null) {
+                synchronized(FileWrapper.class) {
+                    if (fileWrapperInstance == null) {
+                        fileWrapperInstance = new FileWrapper();
+                    }
+                }
+            }
+            return fileWrapperInstance;
         }
```

Comment:
I think these can be replaced by the following, avoiding the need for the test and synchronization (which is part of the reason for using the Singleton model).

        private static FileWrapper fileWrapperInstance = new FileWrapper();
        private static FileWrapper getFileWrapperInstance() {
            return fileWrapperInstance;
        }